False positive of unused-import since v2.15.0

### Bug description

Given the following code:

```python
# pylint: disable=missing-docstring,too-few-public-methods

import abc
import six


class MyClass(six.with_metaclass(abc.ABCMeta)):
    pass
```

### Configuration

_No response_

### Command used

```shell
pylint a.py
```


### Pylint output

```shell
************* Module t
/usr/local/google/home/yileiyang/Downloads/t.py:4:0: W0611: Unused import six (unused-import)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 2.50/10, +5.00)
```


### Expected behavior

No `unused-import` should be raised

### Pylint version

```shell
pylint 2.15.0
astroid 2.12.10
Python 3.9.13 (main, Jun  1 2022, 14:19:10)
[GCC 11.2.0]
```

This cannot be reproduced in v2.14.5.
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/7506.false_positive
+++ b/doc/whatsnew/fragments/7506.false_positive
@@ -1,0 +1,3 @@
+Fix ``unused-import`` false positive for usage of ``six.with_metaclass``.
+
+Closes #7506

--- a/requirements_test.txt
+++ b/requirements_test.txt
@@ -6,6 +6,7 @@ contributors-txt>=1.0.0
 pytest-cov~=4.0
 pytest-profiling~=1.7
 pytest-xdist~=3.2
+six
 # Type packages for mypy
 types-pkg_resources==0.1.3
 tox>=3

--- a/tests/functional/u/unused/unused_import.py
+++ b/tests/functional/u/unused/unused_import.py
@@ -10,6 +10,7 @@ from sys import flags  # [unused-import]
 # +1:[unused-import,unused-import]
 from collections import deque, OrderedDict, Counter
 import re, html.parser  # [unused-import]
+import six
 
 DATA = Counter()
 # pylint: disable=self-assigning-variable
@@ -98,8 +99,10 @@ if TYPE_CHECKING:
     import zoneinfo
 
 
-class WithMetaclass(metaclass=ABCMeta):
-    pass
+class WithMetaclass(six.with_metaclass(ABCMeta)):
+    """Regression test for https://github.com/PyCQA/pylint/issues/7506.
+
+    Requires six."""
 
 
 # Regression test for https://github.com/pylint-dev/pylint/issues/3765

--- a/tests/functional/u/unused/unused_import.txt
+++ b/tests/functional/u/unused/unused_import.txt
@@ -6,9 +6,9 @@ unused-import:8:0:8:21::Unused flags imported from sys:UNDEFINED
 unused-import:11:0:11:51::Unused OrderedDict imported from collections:UNDEFINED
 unused-import:11:0:11:51::Unused deque imported from collections:UNDEFINED
 unused-import:12:0:12:22::Unused import re:UNDEFINED
-unused-import:16:0:16:40::Unused SomeOtherName imported from fake:UNDEFINED
-unused-import:53:0:53:9::Unused import os:UNDEFINED
-unused-import:88:4:88:19::Unused import unittest:UNDEFINED
-unused-import:90:4:90:15::Unused import uuid:UNDEFINED
-unused-import:92:4:92:19::Unused import warnings:UNDEFINED
-unused-import:94:4:94:21::Unused import compileall:UNDEFINED
+unused-import:17:0:17:40::Unused SomeOtherName imported from fake:UNDEFINED
+unused-import:54:0:54:9::Unused import os:UNDEFINED
+unused-import:89:4:89:19::Unused import unittest:UNDEFINED
+unused-import:91:4:91:15::Unused import uuid:UNDEFINED
+unused-import:93:4:93:19::Unused import warnings:UNDEFINED
+unused-import:95:4:95:21::Unused import compileall:UNDEFINED

